### PR TITLE
python3Packages.pymisp: init at 2.4.174

### PR DESCRIPTION
**Pull request by arunabh98**
This PR adds the pymisp package - https://pymisp.readthedocs.io/en/latest/ to the list of python packages.

### CODE REVIEW

File: pkgs/development/python-modules/pymisp/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+buildPythonPackage rec {
+  version = "2.4.174";
+  pname = "pymisp";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "173247f368e78c9069d201621f3d7746b41cc478d76b923e1a4c1e7f29ada287";
+  };
+
+  buildInputs = [ pytest ];
```
Comment (jsoo1) [inline pkgs/development/python-modules/pymisp/default.nix:14]:
```suggestion
  nativeBuildInputs = [ poetry-core ];
  
  checkInputs = [ pytestCheckHook pytest-mock ];
```

File: pkgs/development/python-modules/pymisp/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{ lib, buildPythonPackage, fetchPypi, python3, pytest
```
Comment (jsoo1) [inline pkgs/development/python-modules/pymisp/default.nix:1]:
```suggestion
{ lib, buildPythonPackage, fetchFromGitHub, pythonOlder, pytestCheckHook, pytest-mock, poetry-core
```

File: pkgs/development/python-modules/pymisp/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+, requests, python-dateutil, jsonschema, deprecated
+}:
+
+buildPythonPackage rec {
+  version = "2.4.174";
+  pname = "pymisp";
+
```
Comment (jsoo1) [inline pkgs/development/python-modules/pymisp/default.nix:8]:
```suggestion
  format = "pyproject.toml";
  disabled = pythonOlder "3.8";
```

File: pkgs/development/python-modules/pymisp/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    requests python-dateutil jsonschema deprecated
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+    rm build -rf
+    ${python3.interpreter} nix_run_setup test
+    runHook postCheck
+  '';
+
```
Comment (jsoo1) [inline pkgs/development/python-modules/pymisp/default.nix:26]:
```suggestion
  # "testing optional email feature"
  pytestFlagsArray = [ "--ignore tests/test_emailobject.py" ];

  disabledTests = [
    "test_mimeType"
    "test_default_attributes"
    "test_feed"
    "test_git_vuln_finder"
    "test_obj_default_values"
    "test_object_tag"
    "test_to_dict_json_format"
  ];

```

File: pkgs/development/python-modules/pymisp/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  meta = with lib; {
+    description = "Python API for MISP";
+    homepage =  "https://github.com/MISP/PyMISP";
+    license = [ licenses.bsd2 ];
```
Comment (jsoo1) [inline pkgs/development/python-modules/pymisp/default.nix:30]:
```suggestion
    maintainers = [ maintainers.jsoo1 ];
    platforms = platforms.unix;
    license = [ licenses.bsd2 ];
```

File: pkgs/development/python-modules/pymisp/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+}:
+
+buildPythonPackage rec {
+  version = "2.4.174";
+  pname = "pymisp";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "173247f368e78c9069d201621f3d7746b41cc478d76b923e1a4c1e7f29ada287";
+  };
```
Comment (jsoo1) [inline pkgs/development/python-modules/pymisp/default.nix:12]:
```suggestion
  src = fetchFromGitHub {
    owner = "MISP";
    repo = "PyMISP";
    rev = "v${version}";
    hash = "sha256-q67LDJiwibG1X9QtzYQSP/qSTSXDSHp60gJIa7coq+A=";
  };
```

This makes sure the tests are run (they are removed when published)

File: pkgs/development/python-modules/pymisp/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
+{ lib, buildPythonPackage, fetchPypi, python3, pytest
+, requests, python-dateutil, jsonschema, deprecated
+}:
+
+buildPythonPackage rec {
```
Comment (jsoo1) [inline pkgs/development/python-modules/pymisp/default.nix:5]:
```suggestion
let version = "2.4.174"; in

buildPythonPackage {
  pname = "pymisp";
  inherit version;
```

File: pkgs/development/python-modules/pymisp/default.nix [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  version = "2.4.174";
+  pname = "pymisp";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "173247f368e78c9069d201621f3d7746b41cc478d76b923e1a4c1e7f29ada287";
+  };
+
```
Comment (jsoo1) [inline pkgs/development/python-modules/pymisp/default.nix:13]:
```suggestion
  postPatch = ''
    substituteInPlace pyproject.toml \
      --replace 'jsonschema = "^4.5.1"' 'jsonschema = "*"'
  '';

```

**Review by jsoo1 — CHANGES_REQUESTED** [review]
This is a great first attempt without having built it!

Let's add these changes (I built something quite like this just now, but it will need a separate check). I just want to update your PR title and these changes should do the trick.

**Comment by arunabh98** [comment]
> This is a great first attempt without having built it!
> 
> Let's add these changes (I built something quite like this just now, but it will need a separate check). I just want to update your PR title and these changes should do the trick.

Thanks for proposing the changes. I committed all the proposed changes. Let me know if this looks good and also what would be the best PR title/description.

**Comment by jsoo1** [comment]
> > This is a great first attempt without having built it!
> > Let's add these changes (I built something quite like this just now, but it will need a separate check). I just want to update your PR title and these changes should do the trick.
> 
> Thanks for proposing the changes. I committed all the proposed changes. Let me know if this looks good and also what would be the best PR title/description.

Let's give them a quick test and this should be good!

**Comment by jsoo1** [comment]
can you apply this please, then squash the commits into one? i just built with these changes:
```diff
commit 046686cd8aafbfe02188e03ecf182243eb0fc36d
Author: John Soo <john.soo@arista.com>
Date:   Wed Aug 2 09:02:41 2023 -0700

    fixes

diff --git a/pkgs/development/python-modules/pymisp/default.nix b/pkgs/development/python-modules/pymisp/default.nix
index b7f9347fe0d2..74ebd04d0ca9 100644
--- a/pkgs/development/python-modules/pymisp/default.nix
+++ b/pkgs/development/python-modules/pymisp/default.nix
@@ -7,38 +7,44 @@ let version = "2.4.174"; in
 buildPythonPackage {
   pname = "pymisp";
   inherit version;
-  version = "2.4.174";
-  pname = "pymisp";
-  format = "pyproject.toml";
+  format = "pyproject";
   disabled = pythonOlder "3.8";
+
   src = fetchFromGitHub {
     owner = "MISP";
     repo = "PyMISP";
     rev = "v${version}";
-    hash = "sha256-q67LDJiwibG1X9QtzYQSP/qSTSXDSHp60gJIa7coq+A=";
+    hash = "sha256-OBchUEz+ZK6XNbnFFL6nsCuXniZ0vAfdJLj6LaFC2i0=";
   };
+
+  patches = [ ./pymisp-remove-poetry-group.patch ];
+
   postPatch = ''
     substituteInPlace pyproject.toml \
-      --replace 'jsonschema = "^4.5.1"' 'jsonschema = "*"'
+      --replace 'jsonschema = "^4.18.4"' 'jsonschema = "*"' \
+      --replace 'requests = "^2.31.0"' 'requests = "*"' \
+      --replace 'deprecated = "^1.2.14"' 'deprecated = "*"' \
+      --replace 'publicsuffixlist = {version = "^0.10.0.20230730", optional = true}' ""
   '';
 
   nativeBuildInputs = [ poetry-core ];
-  
+
   checkInputs = [ pytestCheckHook pytest-mock ];
 
   propagatedBuildInputs = [
     requests python-dateutil jsonschema deprecated
   ];
 
-  # "testing optional email feature"
+  # testing optional email feature
   pytestFlagsArray = [ "--ignore tests/test_emailobject.py" ];
 
   disabledTests = [
-    "test_mimeType"
+    "test_mime"
     "test_default_attributes"
     "test_feed"
     "test_git_vuln_finder"
     "test_obj_default_values"
+    "test_object_galaxy"
     "test_object_tag"
     "test_to_dict_json_format"
   ];
@@ -48,6 +54,6 @@ buildPythonPackage {
     homepage =  "https://github.com/MISP/PyMISP";
     maintainers = [ maintainers.jsoo1 ];
     platforms = platforms.unix;
-    license = [ licenses.bsd2 ];
+    license = licenses.bsd2;
   };
 }
diff --git a/pkgs/development/python-modules/pymisp/pymisp-remove-poetry-group.patch b/pkgs/development/python-modules/pymisp/pymisp-remove-poetry-group.patch
new file mode 100644
index 000000000000..0c17d0c358ec
--- /dev/null
+++ b/pkgs/development/python-modules/pymisp/pymisp-remove-poetry-group.patch
@@ -0,0 +1,25 @@
+diff --git a/pyproject.toml b/pyproject.toml
+index 7fc1a65..15548d8 100644
+--- a/pyproject.toml
++++ b/pyproject.toml
+@@ -71,20 +71,6 @@ url = ['pyfaup']
+ email = ['extract_msg', "RTFDE", "oletools"]
+ brotli = ['urllib3']
+ 
+-[tool.poetry.group.dev.dependencies]
+-requests-mock = "^1.11.0"
+-mypy = "^1.4.1"
+-ipython = [
+-    {version = "<8.13.0", python = "<3.9"},
+-    {version = "^8.13.0", python = ">=3.9"}
+-]
+-jupyterlab = "^4.0.3"
+-types-requests = "^2.31.0.2"
+-types-python-dateutil = "^2.8.19.14"
+-types-redis = "^4.6.0.3"
+-types-Flask = "^1.1.6"
+-pytest-cov = "^4.1.0"
+-
+ [build-system]
+ requires = ["poetry_core>=1.1", "setuptools"]
+ build-backend = "poetry.core.masonry.api"
```

**Comment by arunabh98** [comment]
Applied the changes and squashed the commits into one. Thanks @jsoo1.

Let me know if this looks good.

**Review by jsoo1 — APPROVED** [review]

**Pull request merged**